doc/introduction.md

```diff
@@ -0,0 +1,130 @@
+# Introduction
+
+This document provides an introduction of key concepts, details how
+the system boots, including failure modes, and provides links to
+other documents for further study.
+
+## CLI
+
+The command line interface (CLI, see-ell-i) is the traditional way of
+interacting with single network equipment like switches and routers.
+Today users have come to expect more advanced graphical GUIs, like a web
+interface, to manage a device or NETCONF-based tools that allow for
+managing entire fleets of installed equipment.
+
+Nevertheless, when it comes to initial deployment and debugging, it
+is very useful to know how to navigate and use the CLI.
+
+> Proceed to the [CLI User Guide](cli/tutorial.md).
+
+
+## Key Concepts
+
+The two modes in the CLI are the admin-exec and the configure context.
+When logging in to the system, from the console port or SSH, you land in
+admin-exec.  Here you can inspect the system status and do operations to
+debug networking issues, e.g. ping.  You can also enter configure
+context by typing: `configure` followed by commands to `set`, `edit`,
+apply changes using `leave`, or `abort` and return to admin-exec.
+
+> The [CLI Introduction](cli/introduction.md) can be useful to skim
+> through at this point.
+
+The system has several datastores (or files):
+
+ - `factory-config` consists of a set of default configurations, some
+   static and others generated per-device, e.g., a unique hostname and
+   number of ports/interfaces.   This file is generated at boot, if it
+   does not exist, i.e., only on first boot or after factory reset.
+ - `failure-config` is also generated at boot, from the same YANG models
+   as `factory-config`, and holds the system *Fail Secure Mode*
+ - `startup-config` is created from `factory-config` at boot if it does
+   not exist.  It is loaded as the system configuration on each boot.
+ - `running-config` is what is actively running on the system.  If no
+   changes have been made since the system booted, it is the same as
+   `startup-config`.
+ - `candidate-config` is created from `running-config` when entering the
+   configure context.  Any changes made here can be discarded (`abort`,
+   `rollback`) or committed (`commit`, `leave`) to `running-config`.
+
+> See the [Branding & Releases](branding.md) for information on how
+> `factory-config` and `failure-config` can be adapted to different
+> customer requirements.
+
+
+## System Boot
+
+After the system firmware (BIOS or and [boot loader](boot.md) start
+Linux the following happens.  The various failure modes, e.g., missing
+password in VPD, are detailed later in this section.
+
+![System boot flowchart](img/fail-secure.svg)
+
+ 1. Before mounting the `/cfg` and `/var` file systems, which host the
+    read-writable data like `startup-config` and container images, the
+	system checks if it's time for a factory reset.  If so it wipes the
```

Comment:
This reads a bit like the system will do this from time to time, like `fsck` or something :smile:

Maybe something like "..., the system first checks if a factory reset has been requested by the user. If so..."

Reply:
Great catch, fixing!